src/plugins/main.js

```diff
@@ -118,25 +120,44 @@ export default async (context) => {
     }
   }
 
-  const loadAndSetLocale = async locale => {
-    const { loadLanguageAsync } = require('./utils')
+  const loadAndSetLocale = async (newLocale, init = false) => {
+    // Abort if different domains option enabled
+    if (app.i18n.differentDomains) {
+      return
+    }
 
-    let messages
+    // Abort if newLocale did not change
+    if (newLocale === app.i18n.locale) {
+      return
+    }
 
+    const oldLocale = app.i18n.locale
+    app.i18n.beforeLanguageSwitch(oldLocale, newLocale)
```

Comment:
Actually both `beforeLanguageSwitch` and `onLanguageSwitched` should not trigger on initial setting of the locale also as it's not really a "switch" but more a "initialization".